### PR TITLE
PanelChrome: Improve accessibility landmark markup

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

- Makes the PanelChrome a `section` element, hence it is visible as a page landmark
- Changes the title from a h6 to h2 with the Text component

Verify that the button behaviour is the same as without
image

**Why do we need this feature?**

The heading levels do not make sense at the moment

**Who is this feature for?**

All, and especially screen reader users

**Which issue(s) does this PR fix?**:

Related to #73693

**Special notes for your reviewer:**

Although the heading levels will be a bit weird when using rows, this improvement is a quick win. That specific issue may be addressed at a different point.

### CODE REVIEW

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 1]
```diff
@@ -167,34 +168,38 @@ export function PanelChrome({
     <>
       {/* Non collapsible title */}
       {!collapsible && title && (
-        <h6 title={typeof title === 'string' ? title : undefined} className={styles.title}>
-          {title}
-        </h6>
+        <div className={styles.title}>
+          <Text element="h2" variant="h6" title={typeof title === 'string' ? title : undefined}>
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:172]: Could someone provide an example where the `title` attribute is useful here?

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 2]
```diff
@@ -424,13 +429,12 @@ const getStyles = (theme: GrafanaTheme2) => {
...
-      textOverflow: 'ellipsis',
-      overflow: 'hidden',
-      whiteSpace: 'nowrap',
-      fontSize: theme.typography.h6.fontSize,
-      fontWeight: theme.typography.h6.fontWeight,
+      '& > h2': {
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+      },
```
[the remark this answers is not in the document]
Comment (eledobleefe) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:437]: Is this not the same as the `[truncate]` [prop](https://github.com/grafana/grafana/blob/8f5839fe8ff5ae4e5f3f08f055fe8c788b8ed42c/packages/grafana-ui/src/components/Text/Text.tsx#L91)?

[the remark this answers is not in the document]
Comment (tskarhed) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:437]: Yes! Thank you! I didn't know that!

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 3]
```diff
@@ -167,14 +169,16 @@ export function PanelChrome({
...
+          <Text element="h2" variant="h6" truncate title={typeof title === 'string' ? title : undefined}>
+            {title}
+          </Text>
+        </div>
       )}
 
       {/* Collapsible title */}
       {collapsible && (
-        <h6 className={styles.title}>
+        <div className={styles.title}>
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:181]: The better solution here would probably to have the button next to the text, but still have the text clickable. Just having a button wrapping everything causes super confusing behaviour with a screen reader.

For now, this is a section with a label, so it is visible with the landmark.

Comment (tskarhed) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:181]: Still looking into making truncation work with button > h2. The screen reader seems to handle it well at least.

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx [hunk 4]
```diff
@@ -424,13 +434,11 @@ const getStyles = (theme: GrafanaTheme2) => {
...
       padding: theme.spacing(0, padding),
-      textOverflow: 'ellipsis',
-      overflow: 'hidden',
-      whiteSpace: 'nowrap',
-      fontSize: theme.typography.h6.fontSize,
-      fontWeight: theme.typography.h6.fontWeight,
+      minWidth: 0,
+      '& > h2': {
+        minWidth: 0,
+      },
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx:441]: This solves weird cases where overflow isn't recognized. See https://css-tricks.com/flexbox-truncated-text/

**Review by eledobleefe — APPROVED** [review]
LGTM 👌

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx [hunk 5]
```diff
@@ -152,31 +152,35 @@ it('collapses the controlled panel when user clicks on the chevron or the title'
 
   expect(screen.getByText("Panel's Content")).toBeInTheDocument();
 
-  const button = screen.getByText('Default title');
+  const button = screen.getByRole('button', { name: 'Default title' });
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx:155]: Nice 👏

File: packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx [hunk 6]
```diff
@@ -152,31 +152,35 @@ it('collapses the controlled panel when user clicks on the chevron or the title'
...
+    button.parentElement?.parentElement?.parentElement?.nextElementSibling?.id
+  );
 
   fireEvent.click(button);
 
   expect(screen.queryByText("Panel's Content")).not.toBeInTheDocument();
   // aria-controls should be removed when panel is collapsed
   expect(button).not.toHaveAttribute('aria-controlls');
-  expect(button.parentElement?.parentElement?.nextElementSibling?.id).toBe(undefined);
+  expect(button.parentElement?.parentElement?.parentElement?.nextElementSibling?.id).toBe(undefined);
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx:166]: Not a fan of these very specific nested test selectors, maybe there's another way to match these elements? Maybe giving the expanded div a test-id or aria-role if it's appropriate there?

Comment (tskarhed) [reply packages/grafana-ui/src/components/PanelChrome/PanelChrome.test.tsx:166]: Good point, these may break easily otherwise

**Review by Clarity-89 — APPROVED** [review]
🚀

**Pull request merged**